Qgs quick/external widget update #2

sklencar opened the pull request:
Some improvements related to external resource widget and photopanel

- Added back button on PhotoPanel
- Split preview and camera output - preview's dimension depends on captured image and if the image doesn't fill a whole screen, camera output is still visible on the background (fixed by this PR)

Back button looks like this:
![Screenshot_2019-10-09-14-23-48-257_uk co lutraconsulting](https://user-images.githubusercontent.com/6735606/66481458-446ae080-eaa1-11e9-8a89-216cfdbe78ef.png)

Commit:
[QgsQuick]
Added a back button to photopanel

Commit:
Split camera output and preview.
Visible only one of them at the same time

Commit:
Fix of mixing features external resource value
If a feature's value has been changed, it affects also another feature

Commit:
[QgsQuick] Reverted back photo panel opacity
The value is set to default value, keep it for adjustment anyway

wonder-sk commented on src/quickgui/images/ic_back.svg:
```diff
@@ -0,0 +1,10 @@
+	c0.365,0.364,0.547,0.808,0.547,1.328s-0.182,0.938-0.547,1.25L15,15l7.5,7.5c0.365,0.364,0.547,0.808,0.547,1.328
+	s-0.182,0.964-0.547,1.328l-1.719,1.797c-0.363,0.364-0.807,0.547-1.328,0.547c-0.52,0-0.963-0.183-1.328-0.547L7.5,16.328
+	C7.137,15.964,6.953,15.521,6.953,15s0.184-0.964,0.547-1.328L18.125,3.047z"/>
+</svg>
```
Let's replace this image by official material icon

wonder-sk commented on src/quickgui/plugin/qgsquickphotopanel.qml:
```diff
@@ -38,12 +38,21 @@ Drawer {
   property var captureButtonIcon: QgsQuick.Utils.getThemeIcon("ic_camera_alt_border")
   property var confirmButtonIcon: QgsQuick.Utils.getThemeIcon("ic_check_black")
   property var cancelButtonIcon: QgsQuick.Utils.getThemeIcon("ic_clear_black")
+  property var backButtonSource: QgsQuick.Utils.getThemeIcon("ic_back")
```
Let's move the getThemeIcon() calls to qgsquickfeatureformstyling.qml so it's easier to override them when needed...

wonder-sk commented on src/quickgui/plugin/qgsquickphotopanel.qml:
```diff
@@ -139,87 +150,113 @@ Drawer {
+    }
+
+    Rectangle {
+      id: backButton
```
I would suggest to remove this circle - the back arrow is not properly aligned and it does not look great... better to have outline with the icon just like in the case of camera icon. And we will also get rid of the scaleRatio...

wonder-sk commented on src/quickgui/plugin/editor/qgsquickexternalresource.qml:
```diff
@@ -138,6 +138,13 @@ Item {
+      image.source = image.getSource()
+    }
+  }
+
```
I am a bit worried this workaround may not work well - the change of the image source should not be dependent on the item visibility change. There is already image source being changed on 1. onCurrentValueChanged, 2. Component.onCompleted - so let's try to figure out the flow of events so we can have a more robust fix...

wonder-sk commented on src/quickgui/plugin/qgsquickphotopanel.qml:
```diff
@@ -38,12 +38,21 @@ Drawer {
 
   signal confirmButtonClicked(string path, string filename)
 
+  function cancelEvent() {
```
can we rename this method - e.g. discardCapturedImage() would make it more clear that we're not canceling the whole panel, only the image we already took...

wonder-sk commented on src/quickgui/plugin/qgsquickphotopanel.qml:
```diff
@@ -63,6 +72,7 @@ Drawer {
       camera.stop()
       photoPreview.visible = false
     }
+    photoPreview.source = ""
```
I don't fully understand why we need to set photo preview source to empty string here - especially why both when making the item visible and also when hiding it?

Commit:
Udpated icon
Used material design
removed circular background

Commit:
Resized back icon

Commit:
Refactoring
Renamed cancel event to more descriptive name (PhotoPanel)
Removed clearing of preview image source on visibleChanged - live view and preview are now exclusivly visible, preview is not stuck anymore (PhotoPanel)
Delete redundant funtion call in onComplete function external resource widget
